hydromt_wflow/wflow.py

```diff
@@ -3637,3 +3726,27 @@ def clip_forcing(self, crs=4326, **kwargs):
                 self.grid.raster.bounds
             )
             self.set_forcing(ds_forcing)
+
+    def clip_states(self, crs=4326, **kwargs):
+        """Return clippped states for subbasin.
+
+        Returns
+        -------
+        xarray.DataSet
+            Clipped states.
+        """
+        if len(self.states) > 0:
+            self.logger.info("Clipping NetCDF states..")
+            ds_states = xr.merge(self.states.values()).raster.clip_bbox(
+                self.grid.raster.bounds
+            )
+            # Check for reservoirs/lakes presence in the clipped model
+            remove_maps = []
+            if self._MAPS["resareas"] not in self.grid:
+                if "volume_reservoir" in ds_states:
+                    remove_maps.extend(["volume_reservoir"])
+            if self._MAPS["lakeareas"] not in self.grid:
+                if "waterlevel_lake" in ds_states:
+                    remove_maps.extend(["waterlevel_lake"])
+            ds_states = ds_states.drop_vars(remove_maps)
+            self.set_states(ds_states)
```

Comment:
I'm not sure I understand your comment? You mean the xr.merge?
So yes it's because in core it's a dict and not direct xr.Dataset. But I can either merge or go through the dictionnary keys one by one before clipping too.
Part of the method could definitely be in core yes and there is an issie for it to have general clip methods. But anyhow here it's really a couple of lines and the remove_maps part is specific to wflow so I need to also add the function in hydromt_wflow anyway.

Reply:
Alright, good to know!